api/src/main/java/ai/djl/ndarray/BaseNDManager.java

```diff
@@ -270,6 +277,13 @@ public String toString() {
     /** {@inheritDoc} */
     @Override
     public synchronized void attachInternal(String resourceId, AutoCloseable resource) {
+        if (capped.get()) {
+            throw new IllegalStateException("NDManager is capped for addition of resources.");
+        }
+        attachInternalUncapped(resourceId, resource);
```

Comment:
Is creating the additional method of `attachInternalUncapped` just for calling it with managers? It might be better to keep `attachInternal` and `attachInternalUncapped` combined and just add an additional check like `if (capped.get() && !(resource instanceof NDManager))`. Also, can you add a test for capping?

Reply:
With  the return-resources fix in the other pull request, resource can also be an NDArray, therefore that logic